### ISSUE TITLE
value validation types

### ISSUE THREAD

**Issue by Dashed** [omniscientjs/immstruct#71]
Something like this for immstruct structures
https://github.com/tgriesser/state-types-react
https://facebook.github.io/react/docs/reusable-components.html#prop-validation

API sketch:
```js
const blueprint = {
    foo: immstruct.types.string.isRequired
};

immstruct({
    foo: 'bar'
}, blueprint);
```

**Comment by Dashed** [comment]
Any validation violations would be warnings in the console.

**Comment by unindented** [comment]
@Dashed Would this fit your use case? https://github.com/HurricaneJames/react-immutable-proptypes

**Comment by Dashed** [comment]
@unindented Cool! I'll definitely check this out.

**Comment by mikaelbr** [comment]
@Dashed Yeah, the question is. Do we think this is something that Immstruct should do, or is this a React thing?

**Comment by Dashed** [comment]
@mikaelbr I feel this would help debugging a lot. 

I didn't really elaborate on my first post; but I wanted to suggest to 'copy' proptype checking that React does and perform the type check against a Structure instance instead of `this.props`. That way user can use userland proptypes (e.g. `HurricaneJames/react-immutable-proptypes`) meant for React, and as well as React's proptypes.
